datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/ser/ZonedDateTimeSerializer.java

```diff
@@ -92,6 +97,18 @@ public void serialize(ZonedDateTime value, JsonGenerator g, SerializerProvider p
         super.serialize(value, g, provider);
     }
 
+    @Override
+    protected String formatValue(ZonedDateTime value, SerializerProvider provider) {
+        String formatted = super.formatValue(value, provider);
+        // [modules-java8#333]: `@JsonFormat` with pattern should override `SerializationFeature.WRITE_DATES_WITH_ZONE_ID`
+        if (_formatter != null && _shape == JsonFormat.Shape.STRING) {
```

Comment:
Can we though?

Reply:
I am not sure it is possible to programmatically and safely modify formatters?
It might not be more efficient either. But is there a problem appending TZ here -- like it getting appended twice for some formatters?

Reply:
> I am not sure it is possible to programmatically and safely modify formatters?

I thought the same 😅 but asked just in case.

> It might not be more efficient either. But is there a problem appending TZ here -- like it getting appended twice for some formatters?

Yeah, if there is not really much difference then current approach would suffice